core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1243,6 +1248,43 @@ public boolean isNeededDependenciesCompatibleWithInstalledVersion(PluginManager.
             });
         }
 
+        /**
+         * Get the list of incompatible dependencies (if there are any, as determined by isNeededDependenciesCompatibleWithInstalledVersion)
+         *
+         * @since TODO
+         */
+        public List<Plugin> getDependenciesIncompatibleWithInstalledVersion() {
+            return getDependenciesIncompatibleWithInstalledVersion(new PluginManager.MetadataCache());
+        }
+
+        @Restricted(NoExternalUse.class) // table.jelly
+        @SuppressWarnings("unchecked")
+        public List<Plugin> getDependenciesIncompatibleWithInstalledVersion(PluginManager.MetadataCache cache) {
+            return cache.of("getDependenciesIncompatibleWithInstalledVersion:" + name, List.class, () -> {
+                List<Plugin> incompatiblePlugins = new ArrayList<>();
+                for (Plugin p : getNeededDependencies()) {
+                    if (!p.isCompatibleWithInstalledVersion() || !p.isNeededDependenciesCompatibleWithInstalledVersion()) {
```

Comment:
```suggestion
                    if (!p.isCompatibleWithInstalledVersion() || !p.isNeededDependenciesCompatibleWithInstalledVersion(cache)) {
```

Reply:
The code is a modified copy of the methods introduced in https://github.com/jenkinsci/jenkins/pull/3428 , co not passing the cache argument is not a new isssue. However if we pass it, we can increase the hit ratio from 15% to 72%   CC @jglick